gossip/sfc_index.go

```diff
@@ -271,20 +271,39 @@ func (s *Service) processSfc(block *inter.Block, receipts types.Receipts, blockF
 	// Write cheaters
 	for _, stakerID := range cheaters {
 		staker := s.store.GetSfcStaker(stakerID)
-		if !staker.IsCheater {
-			// write into node DB
-			staker.IsCheater = true
-			s.store.SetSfcStaker(stakerID, staker)
-			// write into SFC contract
-			position := sfcpos.Staker(stakerID)
-			statedb.SetState(sfc.ContractAddress, position.IsCheater(), common.BytesToHash([]byte{1}))
+		if staker.HasFork() {
```

Comment:
- this function processSfc is quite long. May you break it into smaller functions?
- comment 271 Write cheaters. But line 274-275 check if HasFork (equivalent to isCheater) then continue. Plz double check the logic or the comment.